### PR TITLE
add fsl directory support

### PR DESCRIPTION
**Pull request by fauna-chase**
ENG-5623
This will allow users to optionally specify a directory within their project root directory that will be used to store all of their fsl files.

### CODE REVIEW

File: src/commands/project/init.ts [hunk 1]
```diff
@@ -64,9 +53,25 @@ export class ProjectInitCommand extends Command {
       );
     }
 
+    let fslDir: string | undefined = await input({
+      message:
+        "What directory would you like to store your fsl files in? (defaults to current directory)",
```
Comment (jdcryans) [inline src/commands/project/init.ts:58]: I'd add an option to create the folder if it doesn't exist already (maybe with some prompt like "directory <dir> doesn't exist, create? Y/n")

File: src/commands/project/init.ts [hunk 2]
```diff
@@ -64,9 +53,25 @@ export class ProjectInitCommand extends Command {
       );
     }
 
+    let fslDir: string | undefined = await input({
```
Comment (jdcryans) [inline src/commands/project/init.ts:56]: Don't forget to update the README

**Review by jdcryans — APPROVED** [review]

File: src/commands/project/init.ts [hunk 3]
```diff
@@ -35,20 +36,8 @@ export class ProjectInitCommand extends Command {
...
-    } else if (!stat.isDirectory()) {
-      this.error(`Expected a directory, found file at ${projectDir}`);
-    } else {
-      try {
-        fs.accessSync(projectDir, fs.constants.W_OK);
-      } catch (e) {
-        this.error(`Cannot write to directory ${projectDir}`);
-      }
+    if (!isWritableDirectory(projectDir)) {
+      this.error(`${projectDir} must be a writeable directory`);
```
Comment (jdcryans) [inline src/commands/project/init.ts:40]: `bin/dev schema pull
Error: The project fsl directory: /Users/jdcryans/tests/cli-project/testfsl must be a writeable directory.
    at Object.error (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/errors/index.js:27:15)
    at PullSchemaCommand.error (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/command.js:139:23)
    at PullSchemaCommand.error (/Users/jdcryans/git/fauna-shell/src/lib/fauna-command.js:49:11)
    at PullSchemaCommand.init (/Users/jdcryans/git/fauna-shell/src/lib/schema-command.ts:51:14)
    at async PullSchemaCommand._run (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/command.js:116:13)
    at async Config.runCommand (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/config/config.js:329:25)
    at async Object.run (/Users/jdcryans/git/fauna-shell/node_modules/@oclif/core/lib/main.js:89:16)`

This is what I get if I delete the folder. Instead we should identify that it's missing.

Comment (macmv) [reply src/commands/project/init.ts:40]: I think updating `isWriteableDirectory` to return `undefined` or a string error might work nicely here, as there's a couple different states the file can be in (missing, is a file, is not writable).

File: src/lib/config/project-config.ts [hunk 4]
```diff
@@ -6,22 +6,29 @@ import { RootConfig, Config, InvalidConfigError } from ".";
...
+    fslDir?: string
+  ) {
     this.stacks = stacks;
     this.defaultStack = defaultStack;
+    this.fslDir = fslDir;
   }
 
   static fromConfig(config: Config): ProjectConfig {
     const defaultStack = config.strOpt("default");
+    const fslDir = config.strOpt("fsl_directory");
```
Comment (macmv) [inline src/lib/config/project-config.ts:31]: I think this should be called `schema_directory`. We call it schema when referring to it by name elsewhere, FSL is just the file extension.

**Review by macmv — COMMENTED** [review]
looking good, just a couple minor things

File: src/commands/project/init.ts (hunk 3 above)
Comment (fauna-chase) [reply src/commands/project/init.ts:40]: wait @jdcryans you get that entire error?
this is what I get
```
fslDir » local-shell schema diff
 ›   Error: The project fsl directory: /Users/chase/fauna/test-schema/fslDir/fsl must be a writeable directory.
```

Comment (fauna-chase) [reply src/commands/project/init.ts:40]: alright updated to be this:
```
fslDir » local-shell schema push                                                                                                                      1 ↵
 ›   Error: The project fsl directory: /Users/chase/fauna/test-schema/fslDir/fsl does not exist.
```

File: src/commands/project/init.ts [hunk 5]
```diff
@@ -64,9 +55,32 @@ export class ProjectInitCommand extends Command {
...
+    } else if (!dirExists(path.join(projectDir, fslDir))) {
+      this.error(
+        `The project fsl directory: ${path.join(
+          projectDir,
+          fslDir
+        )} does not exist.`
+      );
+    } else if (!dirIsWriteable(path.join(projectDir, fslDir))) {
+      this.error(
+        `The project fsl directory: ${path.join(
```
Comment (macmv) [inline src/commands/project/init.ts:73]:
```suggestion
        `The project's schema directory: ${path.join(
```

File: src/commands/project/init.ts [hunk 6]
```diff
@@ -64,9 +55,32 @@ export class ProjectInitCommand extends Command {
...
 
+    let fslDir: string | undefined = await input({
+      message:
+        "What directory would you like to store your fsl files in? (defaults to current directory)",
+    });
+    if (fslDir === "") {
+      fslDir = undefined;
+    } else if (!dirExists(path.join(projectDir, fslDir))) {
+      this.error(
+        `The project fsl directory: ${path.join(
```
Comment (macmv) [inline src/commands/project/init.ts:66]:
```suggestion
        `The project's schema directory: ${path.join(
```

**Review by macmv — APPROVED** [review]
looks good, just comments on error wording

**Pull request merged**